examples/BodySegmentation-maskbodyparts-select/sketch.js

```diff
@@ -0,0 +1,54 @@
+/*
+ * 👋 Hello! This is an ml5.js example made and shared with ❤️.
+ * Learn more about the ml5.js project: https://ml5js.org/
+ * ml5.js license and Code of Conduct: https://github.com/ml5js/ml5-next-gen/blob/main/LICENSE.md
+ *
+ * This example demonstrates segmenting a person by body parts with ml5.bodySegmentation.
+ */
+
+let bodyPix;
+let video;
+let segmentation;
+
+let options = {
+  maskType: "parts",
+};
+
+function preload() {
+  bodyPix = ml5.bodySegmentation("BodyPix", options);
+}
+
+function setup() {
+  createCanvas(640, 480);
+  // Create the video
+  video = createCapture(VIDEO);
+  video.size(width, height);
+  video.hide();
+
+  bodyPix.detectStart(video, gotResults);
+}
+
+function draw() {
+  background(255);
+  image(video, 0, 0);
+  if (segmentation) {
+    // segmentation.values is an array that contains a number for each
+    // pixel of the input image, specifying the body part detected
+    let gridSize = 10;
+    for (let x=0; x < video.width; x += gridSize) {
+      for (let y=0; y < video.height; y += gridSize) {
+        // torsoFront is e.g. 12
+        if (segmentation.values[y * video.width + x] == 12) {
```

Comment:
Will do a more detailed review later, but a quick idea! What about we add some time of `enum` constants as part of `BodySegmentation`? so the code could read:

```js
if (segmentation.values[y * video.width + x] == BodySegmentation.RIGHT_FACE) {
```

or would this make more sense?

```js
if (segmentation.values[y * video.width + x] == ml5.RIGHT_FACE) {
```

@ziyuan-linn is there anything in other models like this as precedent? 

The numbers and names from BodyPix can be in the [tfjs-models](
https://github.com/tensorflow/tfjs-models/tree/master/body-segmentation/src/body_pix#run-inference) repo.

(This is important info for the documentation page! cc @MOQN @QuinnHe @alanvww @myrahsa, sorry for so many tags, not sure who is working on this particular model docs!)

Reply:
Thinking that makes a lot of sense for {read,use}ability.

One caveat would that we'd be baking the internals of a particular model in the API, where even tf's _BodySegmentation_ is trying (albeit not very successfully) to keep the interface agnostic... so maybe rather `BodySegmentation.RIGHT_FACE`, and perhaps those properties only get added with the BodyPix model (and not the SelfieSegmentation one)?

Reply:
Ah yes, that makes sense! Would something like `BodyPix.RIGHT_FACE` help make it more clear? We could also put strings into the array or something instead of integers but that might get tricky to maintain and be prone to error with mispellings, etc.

Reply:
Would this be `BodySegmentation.BodyPix.RIGHT_FACE`? Or `ml5.BodyPix.RIGHT_FACE`? (both seem a bit like a mouthful imho 🤔)

Reply:
Oh, haha, no I think we should *only* do one of these:

```js
BodySegmentation.RIGHT_FACE
ml5.RIGHT_FACE
BodyPix.RIGHT_FACE
```

While there could be other models that have different integer values for similar or the same parts later, I think it's ok to leave that for a problem another day? 

I might opt for `ml5.RIGHT_FACE` it's both the fewest characters and also implies that this is something we are doing above and beyond the model's native behavior itself?

I'm not sure! Open to any and all ideas!

Reply:
I implemented the `BodySegmentation.torsoFront` avenue for testing below, if anyone wants to give it a spin.

The properties currently get added to the instance whenever the BodyPix model is loaded. In practice this means that one would access it through the user-defined name of the instance variable - which could be short, e.g. just `body`. No strong opinions on either on that (also noticing that my JavaScript is quite rusty - please feel welcome to solve this more elegantly).

As for naming: I've been reusing the data from the existing `BODYPIX_PALETTE.js` file, which has it camel-case. The file isn't used for anything else (anymore), so we could easily change it e.g. to `TORSO_FRONT`.

Reply:
Using `enum` constants is a great idea!! I would vote for `BodyPix.RIGHT_FACE`. I will elaborate on it in the following comment.